classes/Customer.php

```diff
@@ -218,8 +218,8 @@ class CustomerCore extends ObjectModel
      */
     public function __construct($id = null)
     {
-        parent::__construct($id);
         $this->id_default_group = (int) Configuration::get('PS_CUSTOMER_GROUP');
+        parent::__construct($id);
```

Comment:
@matks I'm not sure why, but this change was introduced in https://github.com/PrestaShop/PrestaShop/pull/12079/files#diff-83536235df8e0d38c1fdcfd4b2613a0cR222 hope I won't break anything else. :crossed_fingers:

Reply:
Let's ask the dev 😄 
@PierreRambaud is that OK with you ?

Reply:
I think it's @tomlev :smiley:

Reply:
Indeed, but @PierreRambaud merged it 😄 
@tomlev we'd like to revert 1 of your change, please speak before we do it 😏

Reply:
wether we revert it or not we should add a comment to explain the tricky behavior

Reply:
If we revert it, there is no tricky behavior anymore, unless I don't know something about static cache, which I can't document. :/

Reply:
Because @tomlev did it, you can revert everything you want!

Reply:
Well IMHO setting a class field before the parent constructor is not the most common case. Furthermore in this case the order seems to be **very important** so it should be commented to justify the specific order so that we won't revert it back in 6 months ^^

Reply:
Done.